gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/pathfinder/AbstractPathFinder.java

```diff
@@ -103,6 +105,11 @@
       this.destNodes.add(destNode);
     }
 
+    // All dest nodes should be the same class
+    if (this.destNodes != null && this.destNodes.stream().map(Object::getClass).collect(Collectors.toSet()).size() > 1) {
```

Comment:
Can our path finding algorithm return dest nodes with different types? just curious.

Reply:
This is the destination nodes specified by the user in the `destinationIdentifiers` property, it allows more than one. This makes it explicitly fail if they specified an HDFS and mysql cluster for example (which would probably fail somewhere else in compilation anyway).